### PR TITLE
Bug 1702425 - Add tests to Typescript sample and CI job to run them

### PR DESCRIPTION
**Pull request by AmrDhaliwal**
Creating a new pull request after the generated folder was created.

### CODE REVIEW

**Comment by AmrDhaliwal** [comment]
@brizental Hi just made this pull request so the test.ts code can be reviewed.

**Review by brizental — CHANGES_REQUESTED** [review]
Test look good. Please also add a CI job to run it. You can do exactly what we do for the JS sample :) https://github.com/mozilla/glean.js/blob/main/.circleci/config.yml#L56-L80

**Comment by AmrDhaliwal** [comment]
@brizental I added the CI job and I think I broke something.

File: .circleci/config.yml [hunk 1]
```diff
@@ -66,6 +66,33 @@ jobs:
...
+              npm --prefix ./glean run build:webext
+              npm --prefix ./glean run build:cli
+              npm --prefix ./glean link
+      - run:
+          name: Install Typescript dependencies for the TS sample
+          command: |
+              npm --prefix ./samples/web-extension/typescript install
+              npm --prefix ./samples/web-extension/typescript link @mozilla/glean
+      - run:
+          name: Build the JS sample
```
Comment (brizental) [inline .circleci/config.yml:89]:
```suggestion
          name: Build the Typescript sample
```

File: .circleci/config.yml [hunk 2]
```diff
@@ -66,6 +66,33 @@ jobs:
           name: Run samples tests
           command: export PATH=.:$PATH && npm --prefix ./samples/web-extension/javascript run test
 
+samples-tests:
```
Comment (brizental) [inline .circleci/config.yml:69]: This property has the exact same identifier as the one above, this is probably the issue. Maybe don't create a whole new one, but instead add the typecript steps to the `samples-tests` jobe above.

**Review by brizental — CHANGES_REQUESTED** [review]

File: .circleci/config.yml (hunk 2 above)
Comment (AmrDhaliwal) [reply .circleci/config.yml]: I seem to still be failing the ci checks.
Is it still the export PATH?

Comment (AmrDhaliwal) [reply .circleci/config.yml]: Oh I think I fixed it the commands were basically duplicated causing the errors.

Comment (AmrDhaliwal) [reply .circleci/config.yml]: @brizental it looks like the CI job failed since it timed out.

File: .circleci/config.yml [hunk 3]
```diff
@@ -55,16 +55,22 @@ jobs:
...
+          name: Build the JS and TS samples
+          command: |
+            npm --prefix ./samples/web-extension/javascript run build
+            npm --prefix ./samples/web-extension/typescript run build
       - run:
           name: Run samples tests
-          command: export PATH=.:$PATH && npm --prefix ./samples/web-extension/javascript run test
+          command: |
+            export PATH=.:$PATH && npm --prefix ./samples/web-extension/javascript run test
+            export PATH=.:$PATH && npm --prefix ./samples/web-extension/typescript run test
```
Comment (brizental) [inline .circleci/config.yml:73]: Just for us to see the error logs in more contained manner, would you mind moving the typescript command into their own jobs? Something like:

```yaml
name: Install Javascript dependencies for the JS sample
command: |
    npm --prefix ./samples/web-extension/javascript install
    npm --prefix ./samples/web-extension/javascript link @mozilla/glean
<...other JS sample related jobs...>
name: Install Typescript dependencies for the TS sample
    npm --prefix ./samples/web-extension/typescript install
    npm --prefix ./samples/web-extension/typescript link @mozilla/glean
<...other TS sample related jobs...>
```

**Comment by AmrDhaliwal** [comment]
@brizental All the checks have passed please let me know if you need anything else for this bug.

**Review by brizental — APPROVED** [review]
Thanks for sticking to this bug @AmrDhaliwal ! The changes look good, please kindly rebase this branch before we can merge it :) Also, let me know if you are interested in taking a next bug and I can point you to some nice ones.

**Comment by AmrDhaliwal** [comment]
@brizental so this was my first time actually rebasing a branch using git I'm hoping it was correct.

Also, no problem had fun even though it proved more difficult than I thought it would be.
And I would love to start on another bug if you have one in mind.

**Comment by brizental** [comment]
> @brizental so this was my first time actually rebasing a branch using git I'm hoping it was correct.

Seems like you missed some steps. No worries though, try again following this instructions:

Inside the glean.js repository folder:

```bash
# Assuming the remote that points to the mozilla/glean.js repository is called `upstream`, 
# fetch all new changes without applying them:
git fetch upstream 
# Then rebase your branch on top of the newest changes in upstream's main branch:
git rebase upstream/main
```

Github is showing that your branch has conflicts, upon the second step you will probably be warned there are conflicts. Resolve the conflicts, then `git add` the changes files and run the command `git rebase --continue` to proceed with the rebase.

VSCode has [a nice visual  tool](https://code.visualstudio.com/docs/editor/versioncontrol#_merge-conflicts) to help with resolving conflicts, I use that quite a lot. Check it out if you have issues resolving the conflicts too :)

Once you are done with the rebase, use `git log` to check that your history is correct and then finally `git push origin --force-with-lease`. Force is necessary here because with rebase you will essentially change past history of your branch.

**Comment by AmrDhaliwal** [comment]
@brizental when I try to rebase the branch and attempt to commit I keep getting a printout in the console saying "Nothing to commit working tree clean".

**Comment by brizental** [comment]
> @brizental when I try to rebase the branch and attempt to commit I keep getting a printout in the console saying "Nothing to commit working tree clean".

You don't need to create any new commits indeed. Rebasing is about organizing the git history instead of creating new commits :)

**Comment by AmrDhaliwal** [comment]
When I try to "git rebase --continue" I am now receiving the error "no rebase in progress?".

**Comment by AmrDhaliwal** [comment]
OH, these conflicts are the ones with the fork of the project I have with the main repo?

**Comment by AmrDhaliwal** [comment]
@brizental looks like I got it to work :)

**Comment by brizental** [comment]
> @brizental looks like I got it to work :)

Indeed! Great work. I will merge this. 

Regarding next bugs, what do you think about giving [Bug 1712912](https://bugzilla.mozilla.org/show_bug.cgi?id=1712912) a try?

**Comment by AmrDhaliwal** [comment]
@brizental bug seems interesting ill comment on it and take it :)

**Pull request merged**